src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -78,22 +77,26 @@ protected void doGet(HttpServletRequest request,HttpServletResponse response){
 	private void getPublishedCourses(HttpServletRequest request,HttpServletResponse response) {
 		M_log.debug("getPublishedCourses: Called");
 		String term = request.getParameter(TERM);
-	    String termName = request.getParameter(TERM_NAME);
+		String termName = request.getParameter(TERM_NAME);
 		String subAccountUrl = canvasURL+"/api/v1/accounts/1/sub_accounts?recursive=true&per_page=100";
 		String url = canvasURL+"/api/v1/accounts/1/courses?enrollment_term_id="+term+"&published=true&per_page=100";
 		CoursesForSubaccounts cfs =new CoursesForSubaccounts();
-		getSubAccountInfo(subAccountUrl,cfs);
-		getThePublishedCourseList(url,cfs);
-		response.setContentType("text/csv");
-		response.setHeader("Content-Disposition","attachment;filename=PublishedCanvasCoursesWithInstructorsFor"+termName+".csv");
-		try {
-			OutputStream outputStream = response.getOutputStream();
-			String outputResult = generateCSVFile(cfs).toString();
-			outputStream.write(outputResult.getBytes());
-			outputStream.flush();
-			outputStream.close();
-		} catch (IOException e) {
-			M_log.error("Writing the csv file to the has some errror: ",e);
+		getSubAccountInfo(subAccountUrl,cfs,response);
```

Comment:
getSubAccountInfo formats and outputs the error output while the caller does the success formatting.  It would be better / clearer to have all the output done on the same level.  Response is passed to getSubAccountInfo mostly to get the response status code.  It could be stored in the cfs (since it already has error information) and then it doesn't need to be supplied at all.